File: src/main/java/com/uwetrottmann/trakt5/enums/SortBy.java
package com.uwetrottmann.trakt5.enums;

import com.google.gson.annotations.SerializedName;

public enum SortBy {
    @SerializedName("rank")
    RANK,

    @SerializedName("added")
    ADDED,

    @SerializedName("title")
    TITLE,

    @SerializedName("released")
    RELEASED,

    @SerializedName("runtime")
    RUNTIME,

    @SerializedName("popularity")
    POPULARITY,

    @SerializedName("percentage")
    PERCENTAGE,

    @SerializedName("votes")
    VOTES,

    @SerializedName("my_rating")
    MY_RATING,

    @SerializedName("random")
    RANDOM

}

File: src/main/kotlin/com/redissi/trakt/enums/SortBy.kt
package com.redissi.trakt.enums

import java.util.*

enum class SortBy(private val value: String) : TraktEnum {
    RANK("rank"), ADDED("added"), TITLE("title"), RELEASED("released"), RUNTIME("runtime"), POPULARITY("popularity"), PERCENTAGE(
        "percentage"
    ),
    VOTES("votes"), MY_RATING("my_rating"), RANDOM("random");

    companion object {
        private val STRING_MAPPING: MutableMap<String, SortBy> =
            HashMap()

        fun fromValue(value: String): SortBy? {
            return STRING_MAPPING[value.toUpperCase(Locale.ROOT)]
        }

        init {
            for (via in values()) {
                STRING_MAPPING[via.toString().toUpperCase(Locale.ROOT)] = via
            }
        }
    }

    override fun toString(): String {
        return value
    }

}